samples/api/oit_linked_lists/oit_linked_lists.cpp

```diff
@@ -250,9 +250,14 @@ void OITLinkedLists::create_gather_pass_objects(const uint32_t width, const uint
 void OITLinkedLists::create_fragment_resources(const uint32_t width, const uint32_t height)
 {
 	{
-		const VkExtent3D image_extent = {width, height, 1};
-		linked_list_head_image        = std::make_unique<vkb::core::Image>(get_device(), image_extent, VK_FORMAT_R32_UINT, VK_IMAGE_USAGE_STORAGE_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT, VMA_MEMORY_USAGE_GPU_ONLY, VK_SAMPLE_COUNT_1_BIT);
-		linked_list_head_image_view   = std::make_unique<vkb::core::ImageView>(*linked_list_head_image, VK_IMAGE_VIEW_TYPE_2D, VK_FORMAT_R32_UINT);
+		vkb::core::ImageBuilder builder{width, height};
```

Comment:
I don't see why or how this is better than before? I personally prefer the way it was.

Reply:
The current image constructor looks like this:

```c++
	Image(
	    Device const         &device,
	    const VkExtent3D     &extent,
	    VkFormat              format,
	    VkImageUsageFlags     image_usage,
	    VmaMemoryUsage        memory_usage       = VMA_MEMORY_USAGE_AUTO,
	    VkSampleCountFlagBits sample_count       = VK_SAMPLE_COUNT_1_BIT,
	    uint32_t              mip_levels         = 1,
	    uint32_t              array_layers       = 1,
	    VkImageTiling         tiling             = VK_IMAGE_TILING_OPTIMAL,
	    VkImageCreateFlags    flags              = 0,
	    uint32_t              num_queue_families = 0,
	    const uint32_t       *queue_families     = nullptr);
```

Even if you ignore the `Device` argument as being standard to pretty much all the VK resource ctors, that's still a dozen parameters with more than half of them being defaulted.  Most style checkers I know start flagging at anything more than 7 ctor paramers.

if you consider `Image(device, extent, format, VK_IMAGE_USAGE_SAMPLED_BIT, VMA_MEMORY_USAGE_AUTO, VK_SAMPLE_COUNT_1_BIT, 1, 6)`, it's not immediately obvious to a downstream consumer of these examples looking at the code what `1` and `6` mean in this context, or why the sample count is set when it hasn't been in prior image creation code.  It's only by already knowing the layout of the constructor that you can understand these values.

The builder pattern allows...

* reasonable defaults for values that are infrequently changed, such as the tiling or sample count
* access to ALL of the possible values you might want to modify, which the current ctor does not
  * In particular, the `VmaMemoryUsage` flag is the only part of `VmaAllocationCreateInfo` that can be customized through this parameter, while `VmaAllocationCreateFlags`, which is likely something consumers would want to change far more often, is not
* The builder pattern makes it perfectly clear to readers of the examples what the meaning of the values being set are, while constructor param doesn't unless you're looking at the constructor definition.

I have no doubt that you can find places in the code where the builder pattern ends up appearing "just as complex" or even consumes more lines of code as the existing constructor based approach, but IMO, in places like this we're sacrificing brevity for readability and clarity to consumers of the example code as a learning tool.  

Granted I made this change before I added the `build` methods to the builder and the `with_vma_usage` call is almost certainly superfluous, so this could be re-written as

```c++
		linked_list_head_image = vkb::core::ImageBuilder{width, height}
		    .with_format(VK_FORMAT_R32_UINT)
		    .with_usage(VK_IMAGE_USAGE_STORAGE_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT)
		    .build_unique(device);
```

Which would be slightly more compact and avoid the temporary named `builder` variable.  Would that make you happier?